### DISCUSSION TITLE
rauc is not talking to bootloader

### DISCUSSION THREAD

**Discussion by synaption** [rauc/rauc#980] [Q&A] [answered]
I am able to update, but updating does not switch the slot automatically, and mark good/bad/active does not seem to have any effect on ```BOOT_A_LEFT=``` in the boot script.  Slots will switch eventually after rebooting increments the ```BOOT_A_LEFT=``` variable.  Any idea what could be wrong?

**Comment by ejoerns** [comment, marked as answer]
First step would be to see if the environment variables are changed on a manual `rauc status mark-active <slot>` with using `fw_printenv`.

If this gives the expected result, maybe bootloader and userspace have a incompatible idea of where the environment storage is located.

This should be easy to verify by writing a specific variable from userspace and see if it can be seen from the environment within u-boot.

**Comment by synaption** [comment]
Thanks.  That helped a lot.  I was able to figure out that my uboot fragment was basically being ignored for some reason and the env offsets were wrong.
